### ISSUE TITLE
Unexpected asynchronous behavior in Pandas leading to race condition

### ISSUE THREAD

**Issue by justinkterry** [jupyter/notebook#3070]
I was loading several large (100MB+) .csv files into pandas for a deep learning project, had a print statement at the end in an attempt to debug something strange and unrelated, and found the race condition shown with full repo instruction shown in the attatched image, source file for testing here: 
![screenshot from 2017-11-22 01-56-09](https://user-images.githubusercontent.com/19422700/33113892-5578e69c-cf28-11e7-82ab-dbcb6f14fd52.png)

https://drive.google.com/file/d/1xUA0e-b7HrpYB7ByT1GWpmYYZZYOn9-G/view?usp=sharing 

Please tell me if you have any questions

**Comment by takluyver** [comment]
You mean the fact that `done` appears before the last warning? The warnings are sent over `stderr`, while 'done' is on `stdout`. Because the streams are buffered, the order can get mixed up like this. Your code is still being run in the correct order.

We have some plans to make changes in ipykernel to make the output appear more like you'd expect:
https://github.com/jupyter/help/issues/111

**Comment by justinkterry** [comment]
Yes, its that. I don't know what you want to do with this issue with the scattered bug reports everywhere; I'll leave it to you.
